devito/symbolics/search.py

```diff
@@ -46,6 +46,8 @@ def __init__(self, query, mode, deep=False):
         self.deep = deep
 
     def _next(self, expr):
+        if isinstance(expr, int):
```

Comment:
the fact that all of a sudden we need this here, is very suspicious. What's going on?

also, you should use `is_integer`, but we should really try our best to avoid changing this piece of code